src/frameName.cpp

```diff
@@ -20,82 +20,32 @@
 #include <string.h>
 #include "frameName.h"
 #include "vmStructs.h"
+#include "threadNames.h"
+#include "profiler.h"
 
 
 FrameName::FrameName(bool simple, bool dotted, bool use_thread_names) :
     _cache(),
     _simple(simple),
-    _dotted(dotted),
-    _thread_count(0),
-    _threads(NULL) {
+    _dotted(dotted) {
 
     // Require printf to use standard C format regardless of system locale
     _saved_locale = uselocale(newlocale(LC_NUMERIC_MASK, "C", (locale_t)0));
 
-    if (use_thread_names && VMThread::available()) {
-        initThreadMap();
+    if (use_thread_names) {
+        ThreadNames storage = Profiler::_instance.threadNames();
```

Comment:
Doesn't this create a redundant copy of the whole map?

Reply:
The idea was to be sure, that `ThreadNames` instance won't be cleared. 
In fact, if `start` for recently attached agent will be called just after `stop` and before `dump*` we have problem much bigger than loss of `ThreadNames`. Should it be fixed somehow?
However, I think we should copy the whole map in `ThreadNames::getNames()`.

Reply:
I understand why you create a copy, but you create two copies in a row (`_instance.threadNames()` + `storage.getNames()`), which seems redundant.

I didn't get what problem you are talking about. start/stop/dump commands are executed under profiler's lock.